src/object/tilemap.cpp

```diff
@@ -209,8 +213,12 @@ ObjectSettings
 TileMap::get_settings() {
   new_size_x = width;
   new_size_y = height;
+  new_offset_x = 0;
```

Comment:
I don't think you want to reset the new_offset_x and new_offset_y values to 0 here. Otherwise they don't stick.